demisto_sdk/commands/common/tools.py

```diff
@@ -391,47 +392,62 @@ def run_command(command, is_silenced=True, exit_on_error=True, cwd=None):
     return output
 
 
-core_pack_list: Optional[
-    list
-] = None  # Initiated in get_core_pack_list function. Here to create a "cached" core_pack_list
+mp_to_core_packs: Dict[
+    MarketplaceVersions, Set[str]
+] = {}  # Initiated in get_core_packs(). Used for caching.
 
 
 @lru_cache(maxsize=128)
-def get_core_pack_list() -> list:
-    """Getting the core pack list from Github content
+def get_core_packs() -> Dict[MarketplaceVersions, Set[str]]:
+    """Getting the core pack from Github content
 
     Returns:
-        Core pack list
+        A mapping from marketplace versions to their core packs.
     """
-    global core_pack_list
-    if isinstance(core_pack_list, list):
-        return core_pack_list
-    if not is_external_repository():
-        core_pack_list = (
-            get_remote_file(
-                "Tests/Marketplace/core_packs_list.json",
-                git_content_config=GitContentConfig(
-                    repo_name=GitContentConfig.OFFICIAL_CONTENT_REPO_NAME,
-                    git_provider=GitProvider.GitHub,
-                ),
-            )
-            or []
-        )
-        core_pack_list.extend(
-            get_remote_file(
-                "Tests/Marketplace/core_packs_mpv2_list.json",
-                git_content_config=GitContentConfig(
-                    repo_name=GitContentConfig.OFFICIAL_CONTENT_REPO_NAME,
-                    git_provider=GitProvider.GitHub,
-                ),
+    if is_external_repository():
+        return {}  # no core packs in external repos.
+
+    global mp_to_core_packs
+    for mp in MarketplaceVersions:
+        if mp not in mp_to_core_packs:
+            mp_core_packs: Union[list, dict] = (
```

Comment:
Please add a comment explaining the type hint (so people can understand without going through the rest of the code)